### PR TITLE
Allow kvm storage plugin to customize diskdef, add geometry

### PR DESCRIPTION
**Pull request by sureshanaparti**
### Description

This PR allows KVM storage plugin an opportunity to customize the libvirt disk XML.

The new interface method customizeLibvirtDiskDef facilitates plugin-specific config without having to add new methods for each specific thing that may need customizing. Existing plugins should be unaffected because of default implementation in interface.

Additionally, this adds the capability to define geometry in the libvirt disk definition.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by sureshanaparti** [comment]
@blueorangutan package

**Comment by sureshanaparti** [comment]
@blueorangutan test

File: plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java [hunk 1]
```diff
@@ -1091,6 +1108,8 @@ public void setSerial(String serial) {
 
         public LibvirtDiskEncryptDetails getLibvirtDiskEncryptDetails() { return this.encryptDetails; }
 
+        public void setGeometry(DiskGeometry geometry) { this.geometry = geometry; }
```
Comment (DaanHoogland) [inline plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtVMDef.java:1111]: format?

**Review by DaanHoogland — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Comment by borisstoyanov** [comment]
@blueorangutan test matrix

**Comment by borisstoyanov** [comment]
@blueorangutan package

**Comment by borisstoyanov** [comment]
@blueorangutan test alma9 kvm-alma9

**Comment by borisstoyanov** [comment]
@blueorangutan test alma9 kvm-alma9 keepEnv

**Review by borisstoyanov — APPROVED** [review]
LGTM, based on code review and smoketests

**Pull request merged**